### DISCUSSION TITLE
「データフェッチ」の質問

### DISCUSSION THREAD

**Discussion by tokuda109** [vuejs-jp/learn.nuxt.com#84] [ハンズオン Q&A] [answered]
https://github.com/vuejs-jp/learn.nuxt.com/blob/main/content/2.concepts/9.data-fetching/index.md で何か困ったことや質問があれば遠慮なく質問して下さい

**Comment by kyagi-0826** [comment]
Nuxtではfetch APIはあまり推奨せず、useFetch・$fetch・useAsyncDataを推奨しているという認識でいいでしょうか。
別にfetch APIを使用しようと思えば使用できるし、何か理由があって使わない方がいい理由ありますか?

**Reply by nalpan** [reply, marked as answer]
@ubugeeei さん口頭解説の要点をまとめます
- fetch APIは環境依存（ブラウザ、Node.js...）が激しい、Nuxtの機能を使うと差異を吸収でき楽
- $fetchのラッパーがuseAsyncData、useAsyncDataのラッパーがuseFetchとなっている
  - 細かい制御をやるなら$fetch、良くも悪くも細かい制御ができないのがuseFetch
